### PR TITLE
LPS-91946 Errors, successes and validation

### PR DESCRIPTION
**Pull request by andresfulla**

### CODE REVIEW

**Comment by p2kmgcl** [comment]
Just started reviewing :)

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+class SegmentsExperienceSelectorModal extends Component {
+	
+	/**
+	 * @inheritdoc
+	 * @param {object} state
+	 * @return {object}
+	 * @review
+	 */
+	prepareStateForRender(state) {
+		if (!state.condition) {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js]: condition for what? :man_shrugging:

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		const nameInput = this.refs.experienceModal.refs.experienceName;
+		const segmentIdInput = this.refs.experienceModal.refs.experienceSegmentId;
+		if (this.experienceForm && this.experienceForm.onSubmit && nameInput.value) {
+			this.experienceForm.onSubmit(
+				nameInput.value,
+				segmentIdInput.value
+			);
+		}
+		else if (!nameInput.value) {
+			this.setState({
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js]: SF

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+				_requiredNameError: true
+			});
+
+			nameInput.focus();
+		}
+	}
+}
+
+SegmentsExperienceSelectorModal.STATE = {
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js]: jsdocs

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.soy [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+								</label>
+
+								<select
+									class="form-control"
+									id="{$porletPrefix}{$experienceForm.experienceName.inputId}"
+									name="experienceSegmentId"
+									ref="experienceSegmentId"
+								>
+									{foreach $segment in $experienceForm.segmentSelector.segments}
+										{if $segment.segmentsEntryId == $experienceForm.segmentSelector.value }
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.soy]:
```Soy
{let $optionAttributes kind="attributes"}
  value="{$segment.segmentsEntryId}"

  {if $segment.segmentsEntryId == $experienceForm.segmentSelector.value    }
    selected
  {/if}
{/let}

<option {$optionAttributes}>
...
```

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.es.js [hunk 5]
```diff
@@ -154,6 +175,106 @@ class SegmentsExperienceSelector extends Component {
...
+	}
+
+	/**
+	 * Clears the edition with a 500 miliseconds wait,
+	 * so the dismissable alert can complete its animation
+	 * @memberof SegmentsExperienceSelector
+	 * @private
+	 * @review
+	 */
+	_dismissEditionError() {
```
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/SegmentsExperienceSelector.es.js]: el problema aquí es que `_dismissEditionError` se le pasa en Soy al modal.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js (hunk 1 above)
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/js/components/toolbar/segmentsExperiences/modal.es.js]: 💩

**Pull request closed without merging**